bw2data/search/indices.py

```diff
@@ -51,19 +118,78 @@ def add_dataset(self, ds):
         self.add_datasets([ds])
 
     def add_datasets(self, datasets):
-        writer = self.get().writer()
-        for ds in datasets:
-            writer.add_document(**self._format_dataset(ds))
-        writer.commit()
+        all_dataset = list(datasets)
+        with self.db.bind_ctx(MODELS):
+            for chunk_range in range(0, len(datasets), 100):
+                for model in MODELS:
+                    model.insert_many(
+                        [
+                            self._format_dataset(ds)
+                            for ds in all_dataset[chunk_range: chunk_range + 100]
+                        ]
+                    ).execute()
 
     def update_dataset(self, ds):
-        writer = self.get().writer()
-        writer.update_document(**self._format_dataset(ds))
-        writer.commit()
+        with self.db.bind_ctx(MODELS):
+            for model in MODELS:
+                model.delete().where(model.code == ds["code"]).execute()
```

Comment:
The `code` by itself isn't enough to guarantee uniqueness - the code is normally made up by hashing "core" data attributes like name, location, and unit. However, if I make a copy of a given database, these attributes won't change (and this is convenient - I can then easily find the corresponding dataset in the copy). So we need to search for both `code` and `database` attribute values - the combination of these two is guaranteed to be unique.